### PR TITLE
Bring Fourfront deployment strategy in line with CGAP

### PR DESCRIPTION
**Pull request by netsettler**
The code for how to deploy has been debugged in CGAP and needs to now be generalized. I'm doing that in two steps, first just reconciling code against CGAP longhand, and then finally I'll replace all that with code from dcicutils that does the same thing.

* Changes to the code for making `production.ini` and to the templates supporting that.
* A bunch of changes to the health page that were on CGAP for different lengths of time (some new today, some for the last several days).
* Support for `any.ini` though it's not turned on until we add more environment variables to beanstlaks.

Opportunistic:

* Expanded versions used on a number of libraries we depend on because the cgap variation of this code successfully allows more such leeway and I noticed it while diffing.
* There was a problem in the `Makefile` where `make macbuild` did a `make clean` while `make build` did not. I've removed the `make clean` call from the `macbuild` target.
* Added the `update` target to `Makefile`.

### CODE REVIEW

File: deploy/tests/test_generate_production_ini.py [hunk 1]
```diff
@@ -81,39 +105,197 @@ def test_build_ini_file_from_template():
...
+            'PROJECT_VERSION = "%s"' % MOCKED_PROJECT_VERSION,
+        ]
+
+        MockFileStream.reset()
 
-        assert lines == [
+        with mock.patch("subprocess.check_output") as mock_check_output:
+            mock_check_output.side_effect = make_mocked_check_output_for_get_version(simulate_git_command=False)
+            with mock.patch("os.path.exists") as mock_exists:
+                def mocked_exists(filename):
```
Comment (j1z0) [inline deploy/tests/test_generate_production_ini.py:212]: Codacy Issue found: [function already defined line 161](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- deploy/tests/test_generate_production_ini.py  11
- deploy/generate_production_ini.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981&bid=17823188)

File: deploy/tests/test_generate_production_ini.py [hunk 2]
```diff
@@ -81,39 +105,197 @@ def test_build_ini_file_from_template():
...
             'SHHH = "my-secret"',
+            'VERSION = "unknown-version-at-20010203045506000000"',  # We mocked datetime.datetime.now() to get this
+            'PROJECT_VERSION = "%s"' % MOCKED_PROJECT_VERSION,
         ]
+
+        MockFileStream.reset()
+
+        # Uncomment this for debugging...
+        # assert False, "PASSED"
+
```
Comment (j1z0) [inline deploy/tests/test_generate_production_ini.py:243]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981)

File: pyproject.toml [hunk 3]
```diff
@@ -129,6 +129,8 @@ xlwt = "1.2.0"
 [tool.poetry.dev-dependencies]
 # flask only for moto[server]
 flask = ">=1.1.1"
+# Here and elsewhere we pin use of moto 1.3.7.
+# It would help to have a record of why. -kmp 27-Apr-2020
```
Comment (willronchetti) [inline pyproject.toml:133]: The reason is after this version, a bug in moto is introduces involving SQS that breaks our code. We cannot upgrade versions until this bug is fixed, but moto has not had a "true" release in months, so we will have to stick with this version.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- deploy/tests/test_generate_production_ini.py  11
- deploy/generate_production_ini.py  14
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981&bid=17823188)

File: deploy/tests/test_generate_production_ini.py [hunk 4]
```diff
@@ -81,39 +113,193 @@ def test_build_ini_file_from_template():
...
+            'VERSION = "%s"' % MOCKED_LOCAL_GIT_VERSION,
+            'PROJECT_VERSION = "%s"' % MOCKED_PROJECT_VERSION,
         ]
+
+        MockFileStream.reset()
+
+        with mock.patch("subprocess.check_output") as mock_check_output:
+            mock_check_output.side_effect = make_mocked_check_output_for_get_version(simulate_git_command=False)
+            with mock.patch("os.path.exists") as mock_exists:
+                def mocked_exists(filename):
```
Comment (j1z0) [inline deploy/tests/test_generate_production_ini.py:216]: Codacy Issue found: [function already defined line 161](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- deploy/generate_production_ini.py  16
- deploy/tests/test_generate_production_ini.py  11
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981&bid=17823188)

File: deploy/generate_production_ini.py [hunk 5]
```diff
@@ -1,39 +1,198 @@
...
+import json
 import os
+import pkg_resources
+import re
+import subprocess
 import sys
+import toml
 import argparse
 
+from dcicutils.env_utils import (
```
Comment (j1z0) [inline deploy/generate_production_ini.py:17]: Codacy Issue found: [Unused prod_bucket_env imported from dcicutils.env_utils](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981)

File: deploy/tests/test_generate_production_ini.py [hunk 6]
```diff
@@ -81,39 +94,193 @@ def test_build_ini_file_from_template():
...
+            'VERSION = "%s"' % MOCKED_LOCAL_GIT_VERSION,
+            'PROJECT_VERSION = "%s"' % MOCKED_PROJECT_VERSION,
+        ]
+
+        MockFileStream.reset()
+
+        with mock.patch("subprocess.check_output") as mock_check_output:
+            mock_check_output.side_effect = make_mocked_check_output_for_get_version(simulate_git_command=False)
+            with mock.patch("os.path.exists") as mock_exists:
+                def mocked_exists(filename):
```
Comment (j1z0) [inline deploy/tests/test_generate_production_ini.py:197]: Codacy Issue found: [function already defined line 161](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981)

File: pyproject.toml (hunk 3 above)
Comment (netsettler) [reply pyproject.toml:126]: Ah, OK. I can add this info in the file. Such info generally should always be there when pinning.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- deploy/generate_production_ini.py  16
- deploy/tests/test_generate_production_ini.py  11
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981&bid=17823188)

File: deploy/generate_production_ini.py (hunk 5 above)
Comment (j1z0) [inline deploy/generate_production_ini.py:17]: Codacy Issue found: [Unused prod_bucket_env imported from dcicutils.env_utils](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981)

File: deploy/tests/test_generate_production_ini.py (hunk 6 above)
Comment (j1z0) [inline deploy/tests/test_generate_production_ini.py:197]: Codacy Issue found: [function already defined line 161](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- deploy/tests/test_generate_production_ini.py  11
- deploy/generate_production_ini.py  16
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981&bid=17823188)

File: deploy/generate_production_ini.py (hunk 5 above)
Comment (j1z0) [inline deploy/generate_production_ini.py:17]: Codacy Issue found: [Unused prod_bucket_env imported from dcicutils.env_utils](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981)

File: deploy/tests/test_generate_production_ini.py [hunk 7]
```diff
@@ -81,39 +95,288 @@ def test_build_ini_file_from_template():
...
+            'VERSION = "%s"' % MOCKED_LOCAL_GIT_VERSION,
+            'PROJECT_VERSION = "%s"' % MOCKED_PROJECT_VERSION,
         ]
+
+        MockFileStream.reset()
+
+        with mock.patch("subprocess.check_output") as mock_check_output:
+            mock_check_output.side_effect = make_mocked_check_output_for_get_version(simulate_git_command=False)
+            with mock.patch("os.path.exists") as mock_exists:
+                def mocked_exists(filename):
```
Comment (j1z0) [inline deploy/tests/test_generate_production_ini.py:198]: Codacy Issue found: [function already defined line 161](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- deploy/tests/test_generate_production_ini.py  11
- deploy/generate_production_ini.py  16
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981&bid=17823188)

File: deploy/tests/test_generate_production_ini.py [hunk 8]
```diff
@@ -81,39 +110,288 @@ def test_build_ini_file_from_template():
...
```
Comment (j1z0) [inline deploy/tests/test_generate_production_ini.py:213]: Codacy Issue found: [function already defined line 161](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981)

File: deploy/generate_production_ini.py [hunk 9]
```diff
@@ -1,39 +1,203 @@
...
```
Comment (j1z0) [inline deploy/generate_production_ini.py:17]: Codacy Issue found: [Unused prod_bucket_env imported from dcicutils.env_utils](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981)

File: pyproject.toml (hunk 3 above)
Comment (netsettler) [reply pyproject.toml]: This got fixed.

File: deploy/tests/test_generate_production_ini.py (hunk 8 above)
Comment (netsettler) [reply deploy/tests/test_generate_production_ini.py:213]: These are internal functions within tests and this is neither an actual nor a style error.
We really need to fix this diagnostic to ignore internal definitions, especially in test files.

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- deploy/generate_production_ini.py  16
- deploy/tests/test_generate_production_ini.py  11
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981&bid=17823188)

File: deploy/tests/test_generate_production_ini.py (hunk 8 above)
Comment (j1z0) [inline deploy/tests/test_generate_production_ini.py:213]: Codacy Issue found: [function already defined line 161](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981)

File: deploy/generate_production_ini.py (hunk 9 above)
Comment (j1z0) [inline deploy/generate_production_ini.py:17]: Codacy Issue found: [Unused prod_bucket_env imported from dcicutils.env_utils](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981)

**Comment by j1z0** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- deploy/generate_production_ini.py  16
- deploy/tests/test_generate_production_ini.py  11
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981&bid=17823188)

File: deploy/tests/test_generate_production_ini.py (hunk 8 above)
Comment (j1z0) [inline deploy/tests/test_generate_production_ini.py:213]: Codacy Issue found: [function already defined line 161](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5434981)

**Pull request merged**